src/infrastructure/jira/jira-service.ts

```diff
@@ -291,6 +313,31 @@ class JiraService {
 		}
 	};
 
+	private async getStoredIssuePropertyValue<T>(
```

Comment:
**nitpick:** Consider renaming the method to `getIssuePropertyJsonValue` to highlight that it is specific to JSON-encoded values.

For example, while `updateAttachedDesignUrlV2IssueProperty` and `updateIngestedDesignsIssueProperty` read/write JSON values (so, it can use this function),  `setAttachedDesignUrlInIssuePropertiesIfMissing` reads/writes a plain string (so, it cannot use this function).